### PR TITLE
vp - added Accept and Deny buttons and functionality to Pending Requests Table

### LINKED ISSUE #86: Professor can accept or deny requests from this page (modifying the column for status). Status should be pending and only the pending and accepted requests show up on this page. Once the professor modifies the status to denied or completed, it moves to the completed requests page.

### PR DESCRIPTION
**Pull request by vprchlik**
Storybook: https://6734fdad790fe98dc6bf77f1-adgpdqbczi.chromatic.com/?path=/docs/components-pendingrequeststable--docs

For Frontend FEATURE #17. Added buttons to accept and deny pending requests on the pending requests table. Only frontend implementation. Once a request is accepted, the request omits the buttons.

Waiting for Completed Requests Page to be finished by Edward before the part where they can be moved to that page is implemented as there is currently no Completed Requests Page. Otherwise, closes #86.

image

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import HomePage from "main/pages/HomePage";
 import ProfilePage from "main/pages/ProfilePage";
 import AdminUsersPage from "main/pages/AdminUsersPage";
 import AdminRequestsPage from "main/pages/AdminRequestsPage";
+// import PendingRequestsPage from "main/pages/PendingRequestsPage";
```
Comment (ELEXG) [inline frontend/src/App.js:6]: can remove commented out code in App.js - it is implemented in #88

**Review by ELEXG — CHANGES_REQUESTED** [review]
LGTM - just need to remove the commented out code in App.js

File: frontend/src/App.js [hunk 2]
```diff
@@ -50,6 +51,13 @@ function App() {
         {hasRole(currentUser, "ROLE_ADMIN") && (
           <Route exact path="/admin/requests" element={<AdminRequestsPage />} />
         )}
+        {/* {hasRole(currentUser, "ROLE_PROFESSOR") && (
+          <Route
+            exact
+            path="/requests/pending"
+            element={<PendingRequestsPage />}
+          />
+        )} */}
```
Comment (ELEXG) [inline frontend/src/App.js:60]: can remove commented out code in App.js - it is implemented in https://github.com/ucsb-cs156-f24/proj-rec-f24-06/pull/88

**Review by ELEXG — DISMISSED** [review]

**Review by pconrad — APPROVED** [review]
LGTM.. strictly speaking, the storybook should have MSW mocks for the POST endpoints to avoid the error messages that pop up when the Accept and Deny buttons are clicked.

We'll let that go,but we will assign 5 points rather than 10 points because of that.

**Pull request merged**